# OS vs. Theme Preference (Test of Independence)

### Objective:
Analyze UI telemetry data from 350 desktop application users to determine if a user's Operating System dictates their Theme Preference (Dark Mode vs. Light Mode), or if the two categories are mathematically independent. 

**The Telemetry Matrix (Observed Data):**
*   **Windows:** 140 Dark Mode | 60 Light Mode
*   **Mac:** 68 Dark Mode | 32 Light Mode
*   **Linux:** 36 Dark Mode | 14 Light Mode

>Test Used: Chi-Square Test of Independence

>Performed by: Raghav Katyal on 30 September 2026

In [1]:
import numpy as np
from scipy import stats

In [3]:
# The columns are perfectly aligned: [Dark Mode, Light Mode].
user_base = np.array([
    [140,60],
    [68,32],
    [36,14]
])

In [ ]:
#Running the test
chi_stat,p_value,_,_ = stats.chi2_contingency(user_base)

In [7]:
print(f"Chi-Square Score : {chi_stat:.2f}")
print(f"P-value : {p_value:.5f}")

Chi-Square Score : 0.27
P-value : 0.87343


>Observation : A Chi-Square score of 0.27 and p-value score of 0.87 shows no strong evidence against the independence of data ; data passes the independence test. Choice of theme is thus observed as not dependent to OS type.